env0/resource_template.go

```diff
@@ -130,36 +130,43 @@ func templateCreatePayloadFromParameters(d *schema.ResourceData) (client.Templat
 		}
 	}
 	onDeployRetries, hasRetriesOnDeploy := d.GetOk("retries_on_deploy")
+	var onDeploy *client.TemplateRetryOn = nil
+	var onDestroy *client.TemplateRetryOn = nil
 	if hasRetriesOnDeploy {
-		if result.Retry == nil {
-			result.Retry = &client.TemplateRetry{}
-		}
-		result.Retry.OnDeploy = &client.TemplateRetryOn{
+		onDeploy = &client.TemplateRetryOn{
 			Times: onDeployRetries.(int),
 		}
 	}
-	if retryOnDeployOnlyIfMatchesRegex, ok := d.GetOk("retry_on_deploy_only_if_matches_regex"); ok {
+	if retryOnDeployOnlyIfMatchesRegex, ok := d.GetOk("retry_on_deploy_only_when_matches_regex"); ok {
```

Comment:
There was a bug here and below. The correct name is `retry_on_deploy_only_when_matches_regex` and not `retry_on_deploy_only_if_matches_regex`